modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/IndexReindexerBackgroundTaskExecutor.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.search.internal.background.task;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.backgroundtask.BackgroundTask;
+import com.liferay.portal.kernel.backgroundtask.BackgroundTaskExecutor;
+import com.liferay.portal.kernel.backgroundtask.constants.BackgroundTaskConstants;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.search.spi.reindexer.IndexReindexer;
+import com.liferay.portal.search.spi.reindexer.IndexReindexerRegistry;
+
+import java.io.Serializable;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Gustavo Lima
+ */
+@Component(
+	property = "background.task.executor.class.name=com.liferay.portal.search.internal.background.task.IndexReindexerBackgroundTaskExecutor",
+	service = {
+		BackgroundTaskExecutor.class, IndexReindexerBackgroundTaskExecutor.class
+	}
+)
+public class IndexReindexerBackgroundTaskExecutor
```

Comment:
if we follow the naming pattern of the executors, i think this should be ReindexIndexReindexerBackgroundTaskExecutor

Reply:
technically they should probably all end with *ReindexBackgroundTaskExecutor, but its probably not worth trying to rename the others